.github/workflows/autoclose.yml

```diff
@@ -9,14 +9,15 @@ permissions:
   contents: write
 
 jobs:
-  dependabot:
+  close:
     runs-on: ubuntu-latest
-    # If the change comes from a dependency-updating bot
-    if: ${{ github.actor == 'dependabot[bot]' || github.actor == 'renovate[bot]' }}
+    # If the check is called 'build' and it failed and the PR is from either dependabot or renovate
+    # See https://docs.github.com/en/actions/learn-github-actions/expressions for docs on this language.
+    if: ${{ github.event.check_run.conclusion == 'failure' && github.event.check_run.name == 'build' && (startsWith(github.event.check_run.pull_requests[0].head.ref, 'renovate/') || startsWith(github.event.check_run.pull_requests[0].head.ref, 'dependabot/')) }}
```

Comment:
Is it possible that `github.event.check_run.pull_requests[0]` could be an index out of bounds? Does it matter?

Reply:
possibly! there's no types or guarantees given in the docs, just an example object so this is somewhat of a shot in the dark

worst case, I think, is that the autoclose action fails

or in a case that I can't quite see, a real user's PR gets autoclosed

(ok worst worst case is that there's somehow an arbitrary code execution security vuln here that lets an attacker make arbitrary changes to this repo, but if that's possible then it's probably an issue with github actions in general and not this one in specific, as far as I can tell)